src/Symfony/Component/VarDumper/Caster/Caster.php

```diff
@@ -71,6 +71,17 @@ public static function castObject($obj, $class, $hasDebugInfo = false)
 
         if ($a) {
             static $publicProperties = [];
+            if (null === $debugClass) {
+                if (\PHP_VERSION_ID >= 80000) {
```

Comment:
```suggestion
                if (function_exists('get_debug_type')) {
```

This would allow us to use the polyfill if available.

Reply:
But this would prevent optimization by opcache. The fallback code is already similar to what the polyfill would do, so no need to leverage it.